api/v1alpha1/clusterlogforwardertemplate_types.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	loggingv1 "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ClusterLogForwarderTemplateSpec defines the desired state of ClusterLogForwarderTemplate
+type ClusterLogForwarderTemplateSpec struct {
+	TemplateSpec *loggingv1.ClusterLogForwarderSpec `json:"templateSpec"`
```

Comment:
I'm not strongly against this name of "templateSpec" but I still believe it would be better and consistent to follow the convention established by core types which is to utilize "template"  You may even consider looking at OpenShift template type to see what they used.

Reply:
I don't like the name but I'm not sure what else to call it. Here's the goal, maybe we can find a better name:
For each HCP and corresponding guest cluster:

- Guest cluster: User needs a place for their local log forwarding rules (CLF) and their desired api-audit log forwarding rules (HLF)  which will  be executed on the HCP
- Management cluster: the HLO must combine users' HLF and SD's own splunk pipeline into a single CLF on each HCP. The "template" is the SD part of this.
- The management-side CLO reconciles the resulting CLF and does the actual management-side log forwarding for both user (from HLF) and SD (from template).

I'd love to come up with a simpler way to express this, without having multiple resources that are effectively just copies of the CLF, but I'm not sure how.

Reply:
since there is some sort of "you provide something and I provide something" would it be better to call it what it is like: GuestSpec or ManagedClusterSpec or the like to make it more obvious that it is intended for a specific part of a topology and it will be combined with something else?  

> I'd love to come up with a simpler way to express this, without having multiple resources that are effectively just copies of the CLF, but I'm not sure how.

I do not think there is but it allows version control and for the HLF to explicitly control what version of the CLF api it supports.  It would possible for it to reconcile a CLF directly to do what is required assuming:
* CLF CRD is deployed by something
* The CLO is never deployed to the cluster.  
I assume the later is not reasonable